scylla/tests/integration/compression.rs

```diff
@@ -0,0 +1,114 @@
+use crate::utils::{test_with_3_node_cluster, unique_keyspace_name, PerformDDL};
+use scylla::client::session::Session;
+use scylla::client::session_builder::SessionBuilder;
+use scylla::statement::query::Query;
+use scylla::frame::Compression;
+
+use scylla_proxy::{ProxyError, WorkerError, ShardAwareness, RequestRule, Reaction, RequestReaction, Condition, RequestOpcode};
+
+use tokio::sync::mpsc;
+use std::sync::Arc;
+
+async fn test_compression(compression: Option<Compression>, text_size: usize, expected_frame_total_size_range: std::ops::Range<usize>) {
+    let res = test_with_3_node_cluster(ShardAwareness::QueryNode, |proxy_uris, translation_map, mut running_proxy| async move {
+    
+        let request_rule = |tx| {
+            RequestRule(
+                Condition::or(Condition::RequestOpcode(RequestOpcode::Execute), Condition::RequestOpcode(RequestOpcode::Query)),
+                RequestReaction::noop().with_feedback_when_performed(tx),
+            )
+        };
+
+        let (request_tx, mut request_rx) = mpsc::unbounded_channel();
+        for running_node in running_proxy.running_nodes.iter_mut() {
+            running_node.change_request_rules(Some(vec![request_rule(request_tx.clone())]));
+        }
+
+        let session: Session = SessionBuilder::new()
+            .known_node(proxy_uris[0].as_str())
+            .address_translator(Arc::new(translation_map))
+            .compression(compression)
+            .build()
+            .await
+            .unwrap();
+
+        let ks = unique_keyspace_name();
+        session.ddl(format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{'class' : 'NetworkTopologyStrategy', 'replication_factor' : 3}}", ks)).await.unwrap();
+        session.use_keyspace(ks, false).await.unwrap();
+        session
+            .ddl("CREATE TABLE test (k text PRIMARY KEY, t text, i int, f float)")
+            .await
+            .unwrap();
+
+        let q = Query::from("INSERT INTO test (k, t, i, f) VALUES (?, ?, ?, ?)");
+        let large_string = "a".repeat(text_size);
+        session.query_unpaged(q.clone(), ("key", large_string.clone(), 42_i32, 24.03_f32)).await.unwrap();
```

Comment:
:pick: It should be able to pass `&str` as a bind value instead of cloning a `String`.

Reply:
```suggestion
        session.query_unpaged(q.clone(), ("key", large_string.as_str(), 42_i32, 24.03_f32)).await.unwrap();
```

Reply:
👍🏼